Source/WTF/wtf/PlatformHave.h

```diff
@@ -1631,3 +1631,11 @@
 #if HAVE(AVKIT) && !PLATFORM(APPLETV)
 #define HAVE_AVPLAYERLAYERVIEW 1
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
+    || PLATFORM(VISION) \
+    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 100000) \
+    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED >= 170000)
+#define HAVE_NEW_NSURL_PARSING 1
```

Comment:
Using "new" in feature names is an anti-pattern. It will be old in no time, with a very misleading name.

Something like HAVE_NSURL_ENCODING_INVALID_CHARACTERS would seem better.